### ISSUE TITLE
Stack Vars from IDA assigned to incorrect locations

### ISSUE THREAD

**Issue by LioTree** [mahaloz/decomp2dbg#51]
Here is a simple example I tested on decomp2dbg v3.1.3 and ida7.5:

```c
#include <stdio.h>

int main()
{
    int a = 1;
    int b;
    scanf("%d",&b);
    int c = a + b;
    printf("%d\n",c);
    return 0;
}
```
the disassembled result of ida is:

```c
int __cdecl main(int argc, const char **argv, const char **envp)
{
  int v4; // [rsp+Ch] [rbp-14h] BYREF
  int v5; // [rsp+10h] [rbp-10h]
  int v6; // [rsp+14h] [rbp-Ch]
  unsigned __int64 v7; // [rsp+18h] [rbp-8h]

  v7 = __readfsqword(0x28u);
  v5 = 1;
  __isoc99_scanf(&unk_2004, &v4, envp);
  v6 = v4 + v5;
  printf("%d\n", (unsigned int)(v4 + v5));
  return 0;
}
```
when I executed to `__isoc99_scanf(&unk_2004, &v4, envp);`, I tried to show the value of v5, but it is different from `$rbp - 0x10`

![image](https://user-images.githubusercontent.com/12847578/204279268-4af847b7-f3e1-4824-bd5b-3ebf1c819d63.png)

Is this a bug or did I do something wrong?

**Comment by mahaloz** [comment]
Hey @LioTree, thanks for taking the time to make this issue, and thank you for the mini example. I assume it was compiled with the default flags for `gcc` right? I suspect I re-introduced this bug when we I touched the lines here: https://github.com/mahaloz/decomp2dbg/blob/055be6b2001954d00db2d683f20e9b714af75880/decomp2dbg/clients/gdb/gdb_client.py#L129

I'll see if I can get to this bug this weekend. If you end up wanting to look, its those lines I linked above that are causing this. What is probably happening is we are assigning it relative to SP when it should be BP... it may require a new indicator on the server side of IDA.
